### DISCUSSION TITLE
Querying a foreign key with a count subquery

### DISCUSSION THREAD

**Discussion by yairhaimo** [supabase/supabase#1113] [Questions] [answered]
I have two tables: `items` and `users` where each `user` can have multiple `items` and there is a FK from the `items` table to the `users` table.
I have the following query:
```
supabase.from('items').select(`id, info, price, images, user (id, name, avatar, phone)`)
```
I want to add an additional field called `number_of_items` to the `user` subobject that is returned in the query. This field will be a simple `count` of all the items that the user has.
1. Is that possible with the current library?
2. If I create a stored procedure/view, is it possible to receive the data in the same fashion (item properties with a user subobject)?

Thanks!

**Comment by steve-chavez** [comment, marked as answer]
Hey @yairhaimo,

> where each user can have multiple items 
> supabase.from('items').select(`id, info, price, images, user (id, name, avatar, phone)`)

Hm, under that form, wouldn't you have repeated user objects for each item?
Perhaps it'd be better to do it backwards - select users and then join items.

> Is that possible with the current library?

If you do it in the way I suggest, then it should be possible with just the library, like:

```js
supabase
 .from('users')
 .select(`id, name, avatar, phone, number_of_items:items(count), items(id, info, price, images)`)
```

Quirk: `number_of_items` will be an array with a single object. To get the count you'd have to do: `number_of_items[0].count`.
Also when using `count` inside a related table, you can only have that column. Having more columns will give an error.

Let me know if that works for you.

**Reply by yairhaimo** [reply]
Thank you! I will give it a thought and check it out.

**Comment by ManishS6** [comment]
Hi
Any idea why the following is giving a error 400
`        const res = await supabaseClient     `
           `     .from('xyz')                         `
            `    .select('*, consignor:consign(id,consign), consignee:consign(id,consign) ,challan(id, driver)')`
             `   .eq('id',id); `
